fe/fe-core/src/main/java/org/apache/doris/planner/Planner.java

```diff
@@ -229,6 +231,9 @@ public void createPlanFragments(StatementBase statement, Analyzer analyzer, TQue
         if (!ConnectContext.get().getSessionVariable().getRuntimeFilterMode().toUpperCase()
                 .equals(TRuntimeFilterMode.OFF.name())) {
             RuntimeFilterGenerator.generateRuntimeFilters(analyzer, rootFragment.getPlanRoot());
+            if (ConnectContext.get().getSessionVariable().enableStarSchemaRuntimeFilterPolicy) {
+                removeUselessRuntimeFilters(rootFragment.getPlanRoot());
```

Comment:
Why not filter when generating rf, but re-traverse remove.

Reply:
we can only remove the useless runtime filters after all needed infomation are set to the plan tree nodes. Because the runtime filter itself is a valid conjuction, we need this info is set on scan node to make a correct decision. During generating phase, we are not able to see the whole picture. Some valid runtime filter hasn't been set to scan node yet.( like tpch q21 ). So after generating phase, the runtime filter are all set, then we can find the invalid ones and remove them.

( The above note is also added as comments, pls see the new commit )